src/modules/ERC20TokenLimitModule.sol

```diff
@@ -118,27 +92,53 @@ contract ERC20TokenLimitModule is BaseModule, IExecutionHookModule {
         return "erc6900.erc20-token-limit-module.1.0.0";
     }
 
+    /// @notice Update the token limit of a validation
+    /// @param entityId The validation entityId to update
+    /// @param token The token address whose limit will be updated
+    /// @param newLimit The new limit of the token for the validation
+    function updateLimits(uint32 entityId, address token, bool hasLimit, uint256 newLimit) public {
+        if (token == address(0)) {
+            revert ERC20NotAllowed(address(0));
+        }
+        limits[entityId][token][msg.sender] = SpendLimit({hasLimit: hasLimit, limit: newLimit});
+    }
+
     /// @inheritdoc BaseModule
     function supportsInterface(bytes4 interfaceId) public view override(BaseModule, IERC165) returns (bool) {
         return interfaceId == type(IExecutionHookModule).interfaceId || super.supportsInterface(interfaceId);
     }
 
-    function _decrementLimit(uint32 entityId, address token, bytes memory innerCalldata) internal {
+    function _decrementLimitIfApplies(uint32 entityId, address token, bytes memory innerCalldata) internal {
+        SpendLimit storage spendLimit = limits[entityId][token][msg.sender];
+
+        if (!spendLimit.hasLimit) {
+            return;
+        }
+
+        if (innerCalldata.length < 68) {
+            revert InvalidCalldataLength();
+        }
+
         bytes4 selector;
         uint256 spend;
-        assembly {
+        assembly ("memory-safe") {
             selector := mload(add(innerCalldata, 32)) // 0:32 is arr len, 32:36 is selector
             spend := mload(add(innerCalldata, 68)) // 36:68 is recipient, 68:100 is spend
         }
-        if (selector == IERC20.transfer.selector || selector == IERC20.approve.selector) {
-            uint256 limit = limits[entityId][token][msg.sender];
+        if (_isAllowedERC20Function(selector)) {
+            uint256 limit = spendLimit.limit;
             if (spend > limit) {
                 revert ExceededTokenLimit();
             }
-            // solhint-disable-next-line reentrancy
-            limits[entityId][token][msg.sender] = limit - spend;
+            unchecked {
```

Comment:
Do we want to check `spend <= limit` and throw a customized error if not?

Reply:
I believe that's checked above in line 130?